azure-quantum/azure/quantum/job/base_job.py

```diff
@@ -0,0 +1,240 @@
+        :param output_data_format: Output data format, defaults to None
+        :type output_data_format: str, optional
+        :param provider_id: Provider ID, defaults to None
+        :type provider_id: str, optional
+        :param input_params: Input parameters, defaults to None
+        :type input_params: Dict[str, Any], optional
+        :param input_params: Input params for job
+        :type input_params: Dict[str, Any]
+        :return: Azure Quantum Job
+        :rtype: Job
+        """
+        job_id = cls.create_job_id()
+        container_name = f"job-{job_id}"
+        container_uri = cls.create_container(
+            workspace=workspace,
+            container_name=container_name
+        )
+        logger.debug(f"Container URI: {container_uri}")
+        input_data_uri = cls.upload_input_data(
+            data=input_data,
+            input_data_name=input_data_name,
+            encoding=encoding
+        )
+
+        return cls.from_uri(
+            workspace=workspace,
+            job_id=job_id,
+            target=target,
+            input_data_uri=input_data_uri,
+            container_uri=container_uri,
+            name=name,
+            input_data_format=input_data_format,
+            output_data_format=output_data_format,
+            provider_id=provider_id,
+            input_params=input_params
+        )
+
+    @classmethod
+    def from_uri(
+        cls,
+        name: str,
+        workspace: "Workspace",
+        job_id: str,
+        target: str,
+        input_data_uri: str,
+        container_uri: str,
+        input_data_format: str = None,
+        output_data_format: str = None,
+        provider_id: str = None,
+        input_params: Dict[str, Any] = None
+    ) -> "BaseJob":
+        """Create new Job from URI if input data is already uploaded
+        to blob storage
+
+        :param name: Job name
+        :type name: str
+        :param workspace: Azure Quantum workspace to submit the blob to
+        :type workspace: "Workspace"
+        :param job_id: Pre-generated job ID
+        :type job_id: str
+        :param target: Azure Quantum target
+        :type target: str
+        :param input_data_uri: Input data URI
+        :type input_data_uri: str
+        :param container_uri: Container URI
+        :type container_uri: str
+        :param input_data_format: Input data format, defaults to None
+        :type input_data_format: str, optional
+        :param output_data_format: Output data format, defaults to None
+        :type output_data_format: str, optional
+        :param provider_id: Provider ID, defaults to None
+        :type provider_id: str, optional
+        :param input_params: Input parameters, defaults to None
+        :type input_params: Dict[str, Any], optional
+        :return: Job instsance
+        :rtype: Job
+        """
+        if input_params is None:
+            input_params = {"params": {"timeout": DEFAULT_TIMEOUT}}
+        if input_data_format is None:
```

Comment:
this is actually not universal for all targets, so can't be set as default.

Reply:
I'll default to an empty `dict` in that case.